integration_tests/src/wallet_daemon_cli.rs

```diff
@@ -721,3 +767,55 @@ pub async fn get_auth_wallet_daemon_client(world: &TariWorld, wallet_daemon_name
         .get_authed_client()
         .await
 }
+
+async fn get_account_from_name(client: &mut WalletDaemonClient, account_name: String) -> Account {
+    let source_account_name = ComponentAddressOrName::Name(account_name.clone());
+    let AccountGetResponse { account, .. } =
+        client
+            .accounts_get(source_account_name.clone())
+            .await
+            .unwrap_or_else(|e| {
+                panic!(
+                    "Failed to get account with name {}. Error: {:?}",
+                    source_account_name, e
+                )
+            });
+    account
+}
+
+async fn submit_unsigned_tx_and_wait_for_response(
+    client: &mut WalletDaemonClient,
+    tx: UnsignedTransaction,
+    inputs: Vec<SubstateRequirement>,
+    account: Account,
+) -> anyhow::Result<TransactionWaitResultResponse> {
+    let submit_req = TransactionSubmitRequest {
+        transaction: Some(tx),
+        signing_key_index: Some(account.key_index),
+        inputs,
+        override_inputs: false,
+        is_dry_run: false,
+        proof_ids: vec![],
+        // TODO: remove
+        fee_instructions: vec![],
+        instructions: vec![],
+        min_epoch: None,
+        max_epoch: None,
+    };
+
+    let submit_resp = client.submit_transaction(submit_req).await?;
+    let wait_req = TransactionWaitResultRequest {
+        transaction_id: submit_resp.transaction_id,
+        timeout_secs: Some(120),
+    };
+    let resp = client
+        .wait_transaction_result(wait_req)
+        .await
+        .unwrap_or_else(|_| panic!("Waiting for the transaction when calling component failed"));
+
+    if let Some(reason) = resp.result.clone().and_then(|finalize| finalize.reject().cloned()) {
```

Comment:
nit: can use references

```suggestion
    if let Some(reason) = resp.result.as_ref().and_then(|finalize| finalize.reject()) {
```

Reply:
will use in next PR